src/main/java/com/bhoruka/bloodbank/web/CampController.java

```diff
@@ -40,19 +48,54 @@ public CampController(@NonNull final CampService campService) {
      * @return the response with the newly created camp id
      */
     @PostMapping("/create")
-    public CreateCampResponse createCamp(@RequestBody final CreateCampRequest createCampRequest) {
+    public Response<CreateCampResponse> createCamp(@RequestBody final CreateCampRequest createCampRequest) {
         log.info("Received create request : {}", createCampRequest);
 
-        CreateCampResponse response = null;
+        Response<CreateCampResponse> response = null;
         try {
             String campId = campService.createCamp(createCampRequest);
-            response = CreateCampResponse.builder()
-                    .campId(campId)
+            response = Response.<CreateCampResponse>builder()
+                    .data(CreateCampResponse.builder()
+                            .campId(campId)
+                            .build())
                     .description(CAMP_CREATE_SUCCESS_MESSAGE)
                     .status(HttpStatus.OK.value())
                     .build();
         } catch (CampCreationFailedException e) {
-            response = CreateCampResponse.builder()
+            response = Response.<CreateCampResponse>builder()
+                    .status(HttpStatus.INTERNAL_SERVER_ERROR.value())
+                    .errorMessage(e.getMessage())
+                    .build();
+        }
+        return response;
+    }
+
+    /**
+     * Controller endpoint for getting data of camp.
+     * @param getCampRequest the incoming get request
+     * @return the response with the get output values
+     */
+    @GetMapping("/get")
+    public Response<GetCampResponse> getCamp(@RequestBody final GetCampRequest getCampRequest) {
+        log.info("Received get request : {}",getCampRequest);
+
+        Response<GetCampResponse> response = null;
+        try {
+            CampModel getCamp = campService.getCamp(getCampRequest);
+
+            response = Response.<GetCampResponse>builder()
+                    .data(GetCampResponse.builder()
+                            .id(getCamp.getId())
+                            .partnerId(getCamp.getPartnerId())
+                            .dateOfCamp(getCamp.getDateOfCamp())
+                            .expectedNoOfDonors(getCamp.getExpectedNoOfDonor())
+                            .actualNoOfDonors(getCamp.getActualNoOfDonor())
+                            .build())
+                    .description(CAMP_GET_SUCCESS_MESSAGE)
+                    .status(HttpStatus.OK.value())
+                    .build();
+        } catch (GetCampDetailsFailedException e) {
+            response = Response.<GetCampResponse>builder()
                     .status(HttpStatus.INTERNAL_SERVER_ERROR.value())
```

Comment:
Since this is thrown only when the db returns an Optional.empty() , it effectively means that a camp with the given camp id is not present in the db. Hence, the status should be 404 Not found.